specs/access-node-architecture/index.md

```diff
@@ -0,0 +1,79 @@
+
+# Goals
+
+- Completely separate the existing Access Node code into un-staked `Observer Service` and the staked `Access Node`
+- Modularize the `Access API` from `Access Node` into its own process/service, which will be able to run against any `Blockchain Data Service`. We will refer to it as the `Flow API Service` going forward.
+- The `Blockchain Data Service` encapsulates the live synchronization concerns provided through consensus follower and execution state sync. It provides access to that state through the `Protocol API` and the `State Streamer` interfaces.
+- The `State Streamer` interface is a generic component which can allow any consumer, such as the `DPS`, to access the state tracked by the `Blockchain Data Service`.
+
+# Non-goals
+
+- Does not address long term historical access (see additional questions)
+- Does not address operational issues for public node operators (rate-limiting, access control etc)
+
+# Detail level objectives
+
+- The handling of received raw protocol and execution state data is managed in the `Blockchain Data Service`. This state is provided to callers via the `Protocol API` and streamed to consumers of the `State Streamer`.
+- `Blockchain Data Service` state is accessed using `State Streamer Registration API` and via `State Streamer` clients.
+- `DPS` indexes execution state that has been delivered live from the protocol by `State Streamer`. The indexed state is used to answer script execution requests from the `DPS API`.
+
+# Design
+
+![architecture.jpg](images/architecture.jpg)
+
+# In detail
+
+To date the implementation of the Access Node has provided the following: 
+
+- Participates as a *staked* node within the Flow network
+- Participates in the private gossip network to follow protocol state
+- Enforces rate limits and throttling of requests
+- Handling of requests to the Access API from public callers for current live spork and
+    - Performs basic request validation
+    - Delegates script execution to upstream Execution Nodes
+    - Delegates transactions to Collection Nodes for processing by the network
+
+Script execution delegation to ENs is an obvious low hanging fruit limiting Access Node scalability and impacting transaction throughput capacity. In addition, scaling of Access API request handling is constrained by tight coupling of code within Access Node. The introduction of the DPS took the first step towards queryable locally held state that could be leveraged to scale Access Node, initially for protocol state only. Subsequently execution state sync provided the missing piece of the blockchain sync story. 
+
+## Revised ownership and responsibility
+
+By managing all chain synchronization concerns `Blockchain Data Service` provides a unified set of functionality that `Access Node` and `Observer Service` are built on, and which can be used as a standalone component for following trusted chain state more generally.
+
```

Comment:
There are a large chunk of sections missing below this. If you did a PDF export of the doc it will not work because of a bug in Notion. The major sections have the breakdown of role/responsibility per component and should not be left out. Thanks

Reply:
I did the Markdown export... and you're rioght I'm seeing that now. I'll add them back in.